pkg/config/loader.go

```diff
@@ -105,28 +104,21 @@ func (l *Loader) Load() (config *Config, err error) {
 func (l *Loader) mergeFile(file string) error {
 	var err error
 
-	// Read the content of the file:
-	var content []byte
-	content, err = ioutil.ReadFile(file)
-	if err != nil {
-		return err
-	}
-
 	// Parse the YAML inside the file:
 	var decoded data.Config
-	err = yaml.Unmarshal(content, &decoded)
+	err = viper.Unmarshal(&decoded)
 	if err != nil {
 		return err
 	}
 
 	// Merge the configuration data from the file with the existing configuration:
-	if decoded.AWX != nil {
+	if &decoded.AWX != nil {
```

Comment:
I think this should be `decoded`, otherwise you are checking if the address of `decoded.AWX` is `nil`, and it will never be.